### PR TITLE
Remove Flow, add TypeScript

### PR DESCRIPTION
**Pull request by jvhoven**
Dependencies:
- removed unused packages
- updated redux for better type definitions
- removed babel as an intermediary build step, rely on the TypeScript compiler for now
- added and configured @typescript-eslint with prettier to enforce coding styleguide
- added ts-jest for interoperability with Jest and TypeScript (configuration can be found in the package.json)

API:
- added extendable MessageTypes enum

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,32 +24,42 @@
...
-    "babel-cli": "6.24.0",
-    "babel-jest": "19.0.0",
-    "babel-plugin-transform-async-to-generator": "6.22.0",
-    "babel-plugin-transform-flow-strip-types": "6.22.0",
-    "babel-plugin-transform-object-rest-spread": "6.23.0",
-    "babel-plugin-transform-runtime": "6.23.0",
-    "babel-preset-es2015": "6.22.0",
-    "babel-preset-react": "6.23.0",
-    "babel-runtime": "6.23.0",
-    "flow-bin": "0.47.0",
```
Comment (Pouja) [inline package.json:37]: Damn that was a really outdated flow version :P

File: src/models.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+export enum MessageTypes {
+  Error = 'ERROR',
+  Warning = 'WARNING',
+  Success = 'SUCCESS',
+  Info = 'INFO',
+  Apocalypse = 'APOCALYPSE',
+}
+
+export type ExtendableMessageTypes<T = {}> = T | MessageTypes;
```
Comment (Pouja) [inline src/models.ts:11]: What is the reason behind making it extendable?

Comment (jvhoven) [reply src/models.ts:11]: https://github.com/42BV/redux-flash-messages/blob/5d242d7348146854d4ffaa18a08cf507fcc757b7/tests/service.test.ts#L159-L170

https://github.com/42BV/redux-flash-messages/blob/5d242d7348146854d4ffaa18a08cf507fcc757b7/README.md#L189-L205

Comment (Pouja) [reply src/models.ts:11]: Ah my bad, I though you added a new functionality. But extending the message type was already part of this project.

**Review by Pouja — DISMISSED** [review]

**Review by Pouja — APPROVED** [review]

**Pull request merged**